packages/react-query/src/types.ts

```diff
@@ -65,6 +76,11 @@ export type UseQueryResult<
   TError = DefaultError,
 > = UseBaseQueryResult<TData, TError>
 
+export type UseSuspenseQueryResult<
+  TData = unknown,
+  TError = DefaultError,
+> = Omit<QueryObserverSuccessResult<TData, TError>, 'isPlaceholderData'>
```

Comment:
I thought If we don't have `enabled` option and have default `suspense` option, we don't have to use `DefinedUseQueryResult` containing unnecessary `QueryObserverRefetchErrorResult`.

```ts
type DefinedQueryObserverResult<TData = unknown, TError = DefaultError> = QueryObserverRefetchErrorResult<TData, TError> | QueryObserverSuccessResult<TData, TError>;
```

We just need only `QueryObserverSuccessResult` for useSuspenseQuery.
So we could type-narrow `isSuccess: false`, `error`, `isError: true` etc.

```ts
interface QueryObserverRefetchErrorResult<TData = unknown, TError = DefaultError> extends QueryObserverBaseResult<TData, TError> {
    data: TData;
    error: TError;
    isError: true;
    isPending: false;
    isLoadingError: false;
    isRefetchError: true;
    isSuccess: false;
    status: 'error';
}
interface QueryObserverSuccessResult<TData = unknown, TError = DefaultError> extends QueryObserverBaseResult<TData, TError> {
    data: TData;
    error: null;
    isError: false;
    isPending: false;
    isLoadingError: false;
    isRefetchError: false;
    isSuccess: true;
    status: 'success';
}
```